src/DynamoCore/Models/DynamoModel.cs

```diff
@@ -1817,9 +1818,10 @@ private void SetPeriodicEvaluation(WorkspaceModel ws)
           bool forceManualExecutionMode,
           out WorkspaceModel workspace)
         {
-            CustomNodeManager.AddUninitializedCustomNodesInPath(
-                Path.GetDirectoryName(filePath),
-                IsTestMode);
+            CustomNodeManager.AddUninitializedCustomNodesInPath(Path.GetDirectoryName(filePath), IsTestMode);
+
+            var currentHomeSpace = Workspaces.OfType<HomeWorkspaceModel>().FirstOrDefault();
+            currentHomeSpace.UndefineCBNFunctionDefinitions();
```

Comment:
why do we need to undefine func definitions before we ReCompileCodeBlockNodesForFunctionDefinitions ?

Reply:
We need to undefine the function definitions if any in the previous workspace before we open a new workspace.

Reply:
ReCompileCodeBlockForFunctionDefinitions will be called _after_ new function definitions have been compiled once the new workspace has been parsed.